src/main/java/org/opensearch/knn/index/codec/transfer/OffHeapFloatVectorTransfer.java

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.knn.index.codec.transfer;
+
+import org.opensearch.knn.jni.JNICommons;
+
+import java.io.IOException;
+import java.util.List;
+
+/**
+ * Transfer float vectors to off heap memory.
+ */
+public final class OffHeapFloatVectorTransfer extends OffHeapVectorTransfer<float[]> {
+
+    public OffHeapFloatVectorTransfer(int transferLimit) {
+        super(transferLimit);
+    }
+
+    @Override
+    protected long transfer(final List<float[]> vectorsToTransfer, boolean append) throws IOException {
+        return JNICommons.storeVectorData(
+            getVectorAddress(),
+            vectorsToTransfer.toArray(new float[][] {}),
```

Comment:
[Not blocking - can take in another review] Instead of this (which I think will copy), should we transfer one vector at a time?

Reply:
This is a shallow copy, so actual vectors won't be copied. The memory gains might not be much